astropy/io/votable/converters.py

```diff
@@ -674,19 +674,25 @@ def __init__(self, field, config=None, pos=None):
 
         precision = field.precision
         width = field.width
-        format_parts = ['%']
+
+        if precision is None:
+            format_parts = ['{0']
```

Comment:
The old ``%6r`` used to work. I can't find a way tough to make it work with ``repr()``. So two workaround appears to be around: change to a str representation, which seems to work once the few test failures are fixed. The difference should be in whether the quotes are shown or not. The other workaround is to ignore width when no precision is given. (Third option is leave the old style).
I would go with option 1), but as module maintainer the decision is yours.